test/test_dynproc.py

```diff
@@ -164,14 +164,29 @@ def testJoin(self):
         size = MPI.COMM_WORLD.Get_size()
         rank = MPI.COMM_WORLD.Get_rank()
         server = client = address = None
-        # crate server/client sockets
+        host = socket.gethostname()
+        addresses = socket.getaddrinfo(host, None)
+        address_families = [ a[0] for a in addresses if a[1]==socket.SOCK_STREAM ]
+        # if both INET and INET6 are available, don't assume the order
+        # is the same on both server and client. Select INET if available.
+        if socket.AF_INET in address_families:
+            socket_family = socket.AF_INET
+        elif socket.AF_INET6 in address_families:
+            socket_family = socket.AF_INET6
+        elif socket.AF_UNIX in address_families:
```

Comment:
I'm not comfortable with using `AF_UNIX`. For it to work, it requires the MPI processes to run on the same host. While that would be OK for a CI builder, it is not OK in general.

Alternatively, you could add something like this:

```
single_host = len(set(MPI.COMM_WORLD.allgather(host))) == 1
...

elif socket.AF_UNIX in address_families and single_host:

```

Reply:
Moreover, I'm not even sure `AF_UNIX` would actually work in the MPIs out there. I believe they should, but we should check for it. Perhaps we should leave this case for future testing enhancements?

Reply:
I ordered it to prefer INET, or INET6 as second choice. So AF_UNIX would only be activated if INET or INET6 are not available.  It's a fallback case.  The question then is, what would you want the test to do if it's running on a UNIX-only network? Similar in a sense to the "undocumented" case I left in the else clause.  In practice I guess it will never happen.